wsmaster/che-core-api-auth/src/main/java/org/eclipse/che/security/oauth/EmbeddedOAuthAPI.java

```diff
@@ -117,32 +118,39 @@ public Response callback(UriInfo uriInfo, @Nullable List<String> errorValues)
               token));
     } catch (OAuthAuthenticationException e) {
       return Response.temporaryRedirect(
-              URI.create(
-                  getParameter(params, "redirect_after_login")
-                      + String.format("&%s=access_denied", ERROR_QUERY_NAME)))
+              URI.create(getRedirectAfterLoginUrl(params, "access_denied")))
           .build();
     } catch (UnsatisfiedScmPreconditionException | ScmConfigurationPersistenceException e) {
       // Skip exception, the token will be stored in the next request.
       LOG.error(e.getMessage(), e);
+    } catch (ScmCommunicationException e) {
```

Comment:
I don't quite understand why ScmCommunicationException is treated like ssl problem.
Could you explain please?

Reply:
That's a good point, I added a check by error code.